error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/shadowing.rs:6:11
   |
6  | #[ensures(result != 3)] 
   |           ^^^^^^^^^^^
   |
note: the error originates here
  --> $DIR/shadowing.rs:7:1
   |
7  | / fn foo(a: i32, b: i32) -> i32 {
8  | |     if a == 42 {
9  | |         let b = 3;
10 | |         b
...  |
14 | |     }
15 | | }
   | |_^
note: counterexample for "a"
  initial value: 42
  final value:   42
  --> $DIR/shadowing.rs:7:8
   |
7  | fn foo(a: i32, b: i32) -> i32 {
   |        ^
note: counterexample for "b"
  initial value: 0
  final value:   0
  --> $DIR/shadowing.rs:7:16
   |
7  | fn foo(a: i32, b: i32) -> i32 {
   |                ^
note: counterexample for "b"
  final value:   3
  --> $DIR/shadowing.rs:9:13
   |
9  |         let b = 3;
   |             ^
note: counterexample for "a"
  final value:   0
  --> $DIR/shadowing.rs:12:13
   |
12 |         let a = 46;
   |             ^
note: counterexample for result
  final value:   3
  --> $DIR/shadowing.rs:7:27
   |
7  | fn foo(a: i32, b: i32) -> i32 {
   |                           ^^^

error: aborting due to previous error

